tool/tsh/db.go

```diff
@@ -634,14 +645,6 @@ func prepareLocalProxyOptions(arg *localProxyConfig) (localProxyOpts, error) {
 
 	// To set correct MySQL server version DB proxy needs additional protocol.
 	if !arg.localProxyTunnel && arg.routeToDatabase.Protocol == defaults.ProtocolMySQL {
-		if arg.database == nil {
-			var err error
-			arg.database, err = getDatabase(arg.cliConf, arg.teleportClient, arg.routeToDatabase.ServiceName)
-			if err != nil {
-				return localProxyOpts{}, trace.Wrap(err)
-			}
-		}
```

Comment:
Not sure why this block was removed. 
Right now the `prepareLocalProxyOptions` function can be also called from :https://github.com/gravitational/teleport/blob/5e182551b4766ff8abe78deb90e38d56c67f152e/tool/tsh/proxy.go#L350 where the `database` var is nil so the `mySQLVersionToProto` in this case will panic.

Reply:
reverting this

Reply:
I removed it originally because the function was only called in two places and both callers already had the database to pass in.

It's confusing that we had an "argument pack" where all of the arguments were implicitly required except one.
In fact, the "listener" in the args should have been checked for nil as well, since if you trace the flow of that arg we eventually do `listener.Accept()` with no nil check. Instead of trying to reason about which args should be checked for nil, I just refactored the function to have explicit args and made both callers supply all of them. I also reduced the number of args from 7 to 5, since we had some unnecessary args being passed in.